.github/workflows/publish-code.yml

```diff
@@ -0,0 +1,74 @@
+name: Publish release
+
+on:
+  release:
+    types:
+      - published
+
+env:
+  NODE_VERSION: '10.x'   # Node 10 LTS
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Check github.ref starts with 'refs/tags/'
+        if: ${{ !startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          echo Error! github.ref does not start with 'refs/tags'
+          echo github.ref: ${{ github.ref }}
+          exit 1
+      - name: Set version number environment variable
+        env:
+          github_ref: ${{ github.ref }}
+        run: |
+          ref="${github_ref:10}"
+          echo $ref
+          echo "::set-env name=ref::$ref"
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - name: yarn
+        run: yarn
+      - name: yarn docs2
+        working-directory: docs2
+        run: yarn
+      - name: yarn gatsby build
+        working-directory: docs2
+        run: yarn gatsby build
+      - name: Publish docs artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: Documentation
+          path: docs2/public/**
+          if-no-files-found: error
+      - name: Use .NET Core 3.1 SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
+          source-url: https://api.nuget.org/v3/index.json
+        env:
+          NUGET_AUTH_TOKEN: ${{secrets.NUGET_AUTH_TOKEN}}
```

Comment:
I see `  run: dotnet nuget push out/* -k ${{ secrets.NUGET_AUTH_TOKEN }}` in the last line of this script. So why to bother with env variable for token?

Reply:
In my tests for the server project, the `-k` option was not needed in the last line of the script.  For some reason, it is required with this project.

We certainly need to set the source-url here, so I set the auth token in both places to be sure it works correctly.  You will see that this file is pretty much a copy of the build-artifacts workflow, which first prompted this change.

Reply:
I don't think I understand. My question was about why declare an `NUGET_AUTH_TOKEN` environment variable and not use it later. I see that the last line uses NUGET_AUTH_TOKEN **from secrets**.

Reply:
That environment variable setting only applies to that step of the workflow.

Reply:
OK but then how does `actions/setup-dotnet@v1` use `NUGET_AUTH_TOKEN` ? I indeed see its usage in docs https://github.com/actions/setup-dotnet but don't understand what it is for.

Reply:
`setup-dotnet` basically just runs `dotnet nuget add source`, as far as I know.

https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-add-source

Reply:
Let's say this is so, but what does the token have to do with it? Does the script stop working without this variable?

Reply:
You know as much as I do. It’s the recommended pattern here according to the GitHub docs. The real question is why the -k option is necessary, because it shouldn’t be. I know the -k wasn’t needed when I tested the workflows for the server project.

Reply:
OK. Let's assume that despite our misunderstanding some parts of this process, we are still moving in the right direction. We'll figure it out over time.